### PR TITLE
add Golang client library

### PR DESCRIPTION
**Pull request by hengm3467**
## Info
- **Description**: 
Add Golang client library

- **Preview**: 
[ Paste the preview link to the edited page here. ]

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves [ Provide the link to the doc issue here. ]

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/dev/go-client-libraries.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/jackc/pgx/v4"
+)
+
+func main() {
+	connStr := "postgres://root:secret@localhost:4566/dev"
```
Comment (neverchanje) [inline docs/dev/go-client-libraries.md]:
```suggestion
         // Please replace the placeholders with the actual configuration.
	connStr := "postgres://USER:PASSWORD@localhost:4566/DATABASE"
```

File: docs/dev/go-client-libraries.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    return err
+
+```
+## Create a materialized view
+
+The code in this section creates a materialized view `counter` to capture the latest total distance and duration..
+
+```go
+sql := `
+		CREATE MATERIALIZED VIEW counter AS
```
Comment (neverchanje) [inline docs/dev/go-client-libraries.md]: The padding here may not be readable enough.

**Comment by neverchanje** [comment]
Below is the demo generated by GPT4. 

---

In this demo, we'll create a materialized view in RisingWave, a streaming database with Postgres-compatible SQL syntax, using the psql command-line tool. Then, we'll use the Go SDK (with the github.com/jackc/pgx package) to retrieve the data from the materialized view.

First, create a SQL file, create_materialized_view.sql, with the following content to define your materialized view:
```sql
CREATE MATERIALIZED VIEW my_materialized_view AS
SELECT
  column1,
  column2,
  column3
FROM
  my_streaming_table;
```
Replace column1, column2, and column3 with the actual column names from your streaming table, and my_streaming_table with the table name.

Execute the create_materialized_view.sql using the psql command-line tool:
```sh
psql -h your_database_host -U your_user -d your_database -f create_materialized_view.sql
```
Replace your_database_host, your_user, and your_database with the appropriate values for your RisingWave database.

Create a Go program, main.go, to connect to the RisingWave database and retrieve data from the materialized view:
```go
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/jackc/pgx/v4"
)

func main() {
	connString := "postgres://your_user:your_password@your_database_host/your_database?sslmode=disable"
	conn, err := pgx.Connect(context.Background(), connString)
	if err != nil {
		log.Fatalf("Unable to connect to the database: %v", err)
	}
	defer conn.Close(context.Background())

	rows, err := conn.Query(context.Background(), "SELECT * FROM my_materialized_view")
	if err != nil {
		log.Fatalf("Error querying materialized view: %v", err)
	}
	defer rows.Close()

	for rows.Next() {
		var column1, column2, column3 string
		err = rows.Scan(&column1, &column2, &column3)
		if err != nil {
			log.Fatalf("Error scanning row: %v", err)
		}
		fmt.Printf("Column1: %s, Column2: %s, Column3: %s\n", column1, column2, column3)
	}

	if err := rows.Err(); err != nil {
		log.Fatalf("Error with rows: %v", err)
	}
}
```
Replace your_user, your_password, your_database_host, and your_database with the appropriate values for your RisingWave database. Also, change the variables and types in the rows.Scan() function based on the actual columns in your materialized view.

Run the Go program:
```sh
go run main.go
```
This will connect to the RisingWave database, retrieve data from the materialized view, and print the results to the console.

**Comment by neverchanje** [comment]
I like the way it tells the user to specify the right arguments. I oftentimes set them wrong.

> Replace your_database_host, your_user, and your_database with the appropriate values for your RisingWave database.

**Comment by hengm3467** [comment]
@neverchanje I got my input from GPT4 as well. I adapted them to be modular (separate functions) instead of just creating a mv. Other than adding the note about replacing the credentials with users' own credentials, what other changes do you suggest? I wanted the examples to be runnable if people follow along but this requirement might not not that important.

**Comment by neverchanje** [comment]
I suggest using psql to create the materialized view. That's also how I wrote the prompts. By default, GPT will create views and tables using SDK.

File: docs/dev/go-client-libraries.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		var row CounterRow
+		err = rows.Scan(&row.total_distance, &row.total_duration)
+		if err != nil {
+			return nil, err
+		}
+		results = append(results, row) //Handle multiple rows of data
+	}
+
+	return results, rows.Err()
+//You might want to print the results by using fmt.Printf or fmt.PrintLn functions
```
Comment (neverchanje) [inline docs/dev/go-client-libraries.md]: These lines of code are not well-formatted...

**Review by neverchanje — APPROVED** [review]
rest lgtm

**Pull request merged**